docs/mkdocs/documentation/ordered_upgrade.md

```diff
@@ -0,0 +1,71 @@
+   is not running on the node prior to kernel module unloading, and to make sure that the workload
+   is back running on the node once the new kernel module version has been loaded
+2. device plugin: managed by KMM operator, that will also manage the unloading/loading of the device plugin
+   on a node being upgraded
+
+## Upgrade flow
+
+1. in case customer wants to use an ordered upgrade issue, he needs to use:
+   kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>=<module version> label on all the nodes that
+   will be used by the kernel module. <module version> value should be the same as the Version
+   field in the KMM CR. 
+2. customer needs to update the appropriate KMM CR.The following fields need to be updated:
+   - ContainerImage (for appropriate kernel version)
+   - Version
+   the update should be atomic: both ContainerImage and Version fields should be updated simultaneously
+2. customer new to bring down its workload on the node that should be upgraded
+3. customer removes kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name> on the node,
+   which will trigger unloading the current kernel module
+4. customer executes additional upgrading actions on the node (if need be)
+5. customer add kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>=<module version> label on the module.
+   <module version> value should be the new Version field value set in the CR (item 2).
+6. customer restores its workload on the node.
+
+In case customer does not need to to execute additional upgrading actions (point 4), then all he needs to do is just to 
+change the kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>=<module version> label.
+Meaning: steps 3,4 and 5 are unified into one step which means: customer changes kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>
+label value to new <module version>, as the one set in the KMM CR
+
+## Inner implementation
+
+### Components:
+
+1. Dedicated controller that watches for kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>. Once it is notified of a change in that label,
+   it will use inner labels: beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name> and
+   beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name> to initiate the upgrade process
+2. ModuleLoader and DevicePlugin daemonsets now have additional node selector value - beta.kmm.node.kubernetes.io/version-module-loader--<module-namespace>-<module-name>
+   and kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name> accordingly. Manipulation of those labels in the nodes will allow the above mentioned
+   controller to remove/add pods on the node
+3. upon change of the Version field in the Module CR, a new ModuleLoader and DevicePlugin daemonsets will be created with appropriate labels in the NodeSelector field
```

Comment:
```suggestion
1. Dedicated controller that watches for `kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>`. Once it is notified of a change in that label,
   it will use inner labels: `beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name>` and
   `beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>` to initiate the upgrade process
2. ModuleLoader and DevicePlugin daemonsets now have additional node selector value - `beta.kmm.node.kubernetes.io/version-module-loader--<module-namespace>-<module-name>`
   and `kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>` accordingly. Manipulation of those labels in the nodes will allow the above mentioned
   controller to remove/add pods on the node
3. upon change of the Version field in the Module CR, a new ModuleLoader and DevicePlugin daemonsets will be created with appropriate labels in the `nodeSelector` field
```